### PR TITLE
Load levels from Tiled

### PR DESCRIPTION
**Pull request by mystal**
Still a WIP, but sending out a pull request to start the review process.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -242,51 +244,98 @@ impl<'a, 'b> State<'a, 'b> {
...
-        // Waypoints
-        world.create_entity()
-            .with(Waypoint {id: 0})
-            .with(Transform::new(100.0, 300.0))
-            .with(Drawable::Waypoint)
-            .build();
+        // Initialize Grid from Grid layer.
+        let mut grid = Grid::new(map.width, map.height, map.tile_width as f32);
+        // TODO: Don't hard-code the layer. Check the name at least.
+        for (j, row) in map.layers[0].tiles.iter().enumerate() {
```
Comment (Ablexxive) [inline src/main.rs]: In general I'm not a fan of single letter variable names in loops but it's fine for now.

File: src/main.rs [hunk 2]
```diff
@@ -242,51 +244,98 @@ impl<'a, 'b> State<'a, 'b> {
...
+        // Iterate over objects. Create Waypoints, Spawners, and Bases.
+        for object in &map.object_groups[0].objects {
+            let obj_type = if let (true, Some(tileset)) = (object.obj_type.is_empty(), map.get_tileset_by_gid(object.gid)) {
+                // Get default value from tileset.
+                let tile_id = object.gid - tileset.first_gid;
+                let tile = &tileset.tiles[tile_id as usize];
+                if let Some(tile_type) = &tile.tile_type {
+                    tile_type.as_ref()
+                } else {
+                    ""
```
Comment (Ablexxive) [inline src/main.rs]: Is this 'else' branch supposed to have content or return an `""`?

**Comment by Ablexxive** [comment]
Yay! This looks great. Lemme know when the rest of it is pushed and I'll review. Also do we want to add the ability to add towers through tiled too?

**Review by Ablexxive — COMMENTED** [review]
Looks good! Yay for keybindings :D

**Review by Ablexxive — APPROVED** [review]

File: src/main.rs (hunk 1 above)
Comment (mystal) [reply src/main.rs]: That's fair. I basically only use single letter variables for loop indices, and basically only i, j, x, y, or z. Happy to change it, though. Any suggestions?

File: src/main.rs (hunk 2 above)
Comment (mystal) [reply src/main.rs]: Yep! It returns "" as the default if no object type was set.

**Pull request merged**